ckan/tests/legacy/functional/api/test_activity.py

```diff
@@ -466,6 +465,10 @@ def _add_resource(self, package, user):
 
         # Test for the presence of a correct activity detail item.
         details = self.activity_details(activity)
+        details = list(filter(
+            lambda x: x['object_type'] == "Resource",
+            details
+        ))
```

Comment:
Filter is kinda deprecated, use list comprehensions. Tends to perform better as since you're removing multiple function calls.

`details = [x for x in details if x['object_type'] == 'Resources']`